plugins/org.eclipse.reddeer.jface/src/org/eclipse/reddeer/jface/dialogs/OkCancelDialog.java

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Red Hat, Inc and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.reddeer.jface.dialogs;
+
+import org.eclipse.reddeer.core.lookup.ShellLookup;
+import org.eclipse.reddeer.swt.impl.button.CancelButton;
+import org.eclipse.reddeer.swt.impl.button.OkButton;
+import org.eclipse.reddeer.swt.impl.shell.DefaultShell;
+import org.eclipse.swt.widgets.Shell;
+import org.hamcrest.Matcher;
+
+/**
+ * Represents a simple dialog with "OK" and "Cancel" buttons.
+ * 
+ * @author lvalach
+ *
+ */
+public class OkCancelDialog extends DefaultShell {
```

Comment:
OkCancelDialog suggests that this represents jface.Dialog class but you are extending DefaultShell (represents swt.Shell) so it should be renamed to OkCancelShell

Reply:
Should I move it also to another package? E.g. org.eclipse.reddeer.swt.impl.shell ?

Reply:
Why are you not using TitleAreaDialog as we talked yesterday? Not a good solution?

Reply:
@apodhrad told me that I should extend DefaultShell. If I remember, you also wasn't satisfied with AbstractWindow, you suggested to extend some ***Dialog,  didn't you? It doesn't matter to me, I will do it as you want.

Reply:
@luvalach I did not notice the package :) 
INHO there are two viable options:
1. OkCancelDialog extends AbstractWindow and exists in o.e.r.jface.dialogs package
2. OkCancelShell extends DefaultShell and exists in o.e.r.swt.impl.shell package

@odockal TitleAreaDialog is not a good parent since a simple OK/Cancel dialog does not have to  have Title Area.

Reply:
Yes, exactly, TitleAreaDialog was the class. At first you wanted to make it abstract, extending AbstractWindow and that you are right, I did not like the idea, but afterwards, if you can extend TitleAreaDialog which is not abstract, one could use your OKCancelDialog directly for some his concrete use case and keeping eclipse's hiearchy, ie: NewCredentialUserDialog extends org.eclipse.jface.dialogs.TitleAreaDialog which is Dialog with ok and cancel buttons... whyt @apodhrad  @rawagner ?